string status
int32 count
---
bool finish